core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -492,6 +492,18 @@ object ZIOSpec extends ZIOBaseSpec {
         } yield assertCompletes
       }
     ),
+    suite("fromFutureInterrupt")(
+      testM("running Future can be interrupted") {
+        for {
+          p1     <- ZIO.effectTotal(scala.concurrent.Promise[Unit])
+          p2     <- ZIO.effectTotal(scala.concurrent.Promise[Unit])
+          fiber  <- ZIO.fromFutureInterrupt(ec => p1.future.map(p2.success)(ec)).fork
+          _      <- fiber.interrupt
+          _      <- ZIO.effectTotal(p1.success(()))
+          result <- ZIO.effectTotal(p2.isCompleted)
```

Comment:
I think the interruption is guaranteed to occur before `p2` can complete. `fiber.interrupt` is not going to return until the finalizer of the interrupted fiber has completed, which means that `interrupted` will have been set to `true` before we begin evaluating `p1.success(())`. Then we complete `p1` and evaluate the future continuation but since `interrupted` is true the continuation will never be submitted for execution.

Reply:
Oh right! I missed that `p1` is only resolved after the interruption.